ui-performance/index.md

```diff
@@ -0,0 +1,484 @@
+setting the `showPerformanceOverlay` property to `true` on the MaterialApp
+or WidgetsApp constructor:
+
+<!-- skip -->
+{% prettify dart %}
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      [[highlight]]showPerformanceOverlay: true,[[/highlight]]
+      title: 'My Awesome App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyHomePage(title: 'My Awesome App'),
+    );
+  }
+}
+{% endprettify %}
+
+You are probably familiar with the Flutter Gallery example app.
+To use the performance overlay with Flutter Gallery,
+[download](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery),
+and run the app in profile mode. The program is written
+so that the app menu allows you to dynamically toggle the overlay,
+as well as enable checks for calls to `saveLayer` and the presence of
+cached images.
+
+<aside class="alert alert-info" markdown="1">
+**Note:** You can’t enable the performance overlay in the Flutter
+Gallery app downloaded from the App Store. That version of the app
+is compiled to release mode (not profile mode), and doesn’t provide
+a menu for enabling or disabling the overlay.
+</aside>
+
+### Identifying problems in the engine graph
+
+If the performance overlay shows red in the engine graph, start by profiling
+the engine (Dart VM), even if the rasterizer graph also shows red.
+To do this, use
+[Observatory](https://dart-lang.github.io/observatory/), Dart’s profiling tool.
+
+#### Displaying Observatory
+
+Observatory provides a statement-level single-stepping debugger for Dart and
+Flutter applications.  It also supports profiling, examining the heap,
+and reporting on code coverage. Observatory's _timeline_ view allows you
+to capture a snapshot of the stack at a moment in time.
+When you open Observatorty's timeline from the Flutter Inspector,
+you'll be using a version that has been customized for Flutter apps.
+
+{% comment %}
+I've talked to tooling SWEs who feel we're not ready to document this.
+{% endcomment %}
+
+Go to Flutter's timeline view in a browser as follows:
+
+<ol markdown="1">
+<li markdown="1">
+Open Observatory using the line chart icon (<img src="images/observatory-timeline-icon.png" alt="zig-zag line chart icon">)
+from the Flutter Inspector.
+
+(Instead, you could open Observatory using the stopwatch icon (<img src="images/observatory-icon.png">),
+but the "view <u>inspector</u>" link takes you to the standard version of the
+timeline, not the version customized for Flutter.)
+
+![IntelliJ Flutter Inspector Window](/images/intellij/visual-debugging.png)
+</li>
+
+<li markdown="1">
+In VS Code, bring up the command palette and enter "observatory".
+Select **Flutter: Open Observatory Timeline** from the list that pops up.
+If this command isn’t available, make sure that the app is running.
+{% comment %}
+[PENDING: Can you go to the timeline directly? If so, which command?]
```

Comment:
/cc @DanTup, we should add an explicit command to link to <observatory>/#/timeline-dashboard.

Reply:
We have :) It's called `Open Observatory Timeline` in the palette though only appears when you're debugging a Flutter app (same as others).

Reply:
At least, that's what it's [supposed to do](https://github.com/Dart-Code/Dart-Code/blob/adfd7fb831dfcbe37dee6d98ae825f8754b19e3b/src/commands/debug.ts#L117-L122). LMK if anyone thinks it's not working right so we can investigate!